airflow/jobs/triggerer_job_runner.py

```diff
@@ -609,6 +609,7 @@ async def run_trigger(self, trigger_id, trigger):
                 self.log.info("Trigger %s fired: %s", self.triggers[trigger_id]["name"], event)
                 self.triggers[trigger_id]["events"] += 1
                 self.events.append((trigger_id, event))
+                break
```

Comment:
I can imagine at least in principle it being desirable to yield event before doing some cleanup work in the trigger itself.  But ...  shortly after the trigger yields first event, it will be unassigned from the task and then shortly after that it will be canceled and it will simply stop running.  So it's probably not a good idea for a trigger to be designed to do anything after the first event.

Reply:
> Won't this result in trigger.cleanup() being called before the trigger finishes though?

Yes, it will be possible, and there won't be any issues with it. Even if the trigger fails because of executing the cleanup process (for ex: deleting some resources used the run method), the triggerer job will ignore the failure and proceed to process only the first yielded event.